src/gleam/result.gleam

```diff
@@ -366,6 +366,29 @@ pub fn all(results: List(Result(a, e))) -> Result(List(a), e) {
   list.try_map(results, fn(x) { x })
 }
 
+/// Given a list of results, returns a pair where the first element is a list
+/// of all the values inside `Ok` and the second element is a list with all the
+/// values inside `Error`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > partition([Ok(1), Error("a"), Error("b"), Ok(2)])
+/// #([1, 2], ["a", "b"])
+/// ```
+///
+pub fn partition(results: List(Result(a, e))) -> #(List(a), List(e)) {
+  do_partition(results, [], [])
+}
+
+fn do_partition(results: List(Result(a, e)), oks: List(a), errors: List(e)) {
+  case results {
+    [] -> #(list.reverse(oks), list.reverse(errors))
```

Comment:
I think we can omit the reversing as this function already discards the original structure of the list. It's faster not to and the user can reverse them if their use case requires it.

Reply:
Sure makes sense 👍 Should the fact that the elements are in reverse order be mentioned in the function doc?
Something like:
```
/// Given a list of results, returns a pair where the first element is a list
/// of all the values inside `Ok` and the second element is a list with all the
/// values inside `Error`. The values in both lists appear in reverse order with
/// respect to their position in the original list of results. 
```

Reply:
Sounds good!